src/client/mod.rs

```diff
@@ -307,8 +307,17 @@ impl Client {
     pub async fn watch(
         &self,
         pipeline: impl IntoIterator<Item = Document>,
-        options: Option<ChangeStreamOptions>,
+        mut options: Option<ChangeStreamOptions>,
     ) -> Result<ChangeStream> {
-        todo!();
+        let pipeline: Vec<Document> = pipeline.into_iter().collect();
+        resolve_options!(self, options, [selection_criteria]);
+
+        let watch = Watch::new(
+            ChangeStreamTarget::Cluster("admin".to_string()),
+            pipeline.clone(),
+            options.clone(),
+        )?;
+        let (change_stream_spec, session) = self.execute_cursor_operation(watch).await?;
+        Ok(ChangeStream::new(change_stream_spec, session, self.clone()))
```

Comment:
given that this implementation is identical across the three `watch` implementations, I think we should abstract it to a single helper that can be called with the various options and an appropriate value for `ChangeStreamTarget`.

Reply:
fixed